pkg/kubelet/kubelet.go

```diff
@@ -1739,8 +1749,8 @@ func (kl *Kubelet) syncPod(pod *api.Pod, mirrorPod *api.Pod, podStatus *kubecont
 	}
 	kl.statusManager.SetPodStatus(pod, apiPodStatus)
 
-	// Kill pods we can't run.
-	if err := canRunPod(pod); err != nil || pod.DeletionTimestamp != nil {
+	// Kill pods we can't run, pods pending deletion, or pods that have been marked as failed.
```

Comment:
The race condition should be avoid if possible as it may lead to confusions, but yes it should not affect correctness.